src/Pact/Types/KeySet.hs

```diff
@@ -114,29 +127,106 @@ instance Arbitrary KeySet where
 -- | allow `{ "keys": [...], "pred": "..." }`, `{ "keys": [...] }`, and just `[...]`,
 -- | the latter cases defaulting to "keys-all"
 instance FromJSON KeySet where
-    parseJSON v = withObject "KeySet" (\o ->
-                    KeySet <$> o .: "keys" <*>
-                    (fromMaybe defPred <$> o .:? "pred")) v <|>
-                  (KeySet <$> parseJSON v <*> pure defPred)
-      where defPred = Name (BareName "keys-all" def)
-instance ToJSON KeySet where
-    toJSON (KeySet k f) = object ["keys" .= k, "pred" .= f]
+    parseJSON v = withObject "KeySet" keyListPred v <|> keyListOnly
+      where
+        defPred = Name (BareName "keys-all" def)
 
+        keyListPred o = KeySet
+          <$> o .: "keys"
+          <*> (fromMaybe defPred <$> o .:? "pred")
+
+        keyListOnly = KeySet
+          <$> parseJSON v
+          <*> pure defPred
+
+instance ToJSON KeySet where
+    toJSON (KeySet k f) = object
+      [ "keys" .= k
+      , "pred" .= f
+      ]
 
 -- -------------------------------------------------------------------------- --
 -- KeySetName
 
-newtype KeySetName = KeySetName Text
-    deriving (Eq,Ord,IsString,AsString,ToJSON,FromJSON,Show,NFData,Generic,SizeOf)
+data KeySetName
+  = KeySetName
+  { _ksnName :: Text
+  , _ksnNamespace :: Maybe NamespaceName
+  } deriving (Eq, Ord, Show, Generic)
+
+instance IsString KeySetName where
+  fromString ksn = case parseAnyKeysetName (T.pack ksn) of
```

Comment:
Is this instance only for tests? This is such a strange instance.

_really_ need to check all usages of `fromString` with this.

Reply:
We can actually do without this and I"d be happy. `IsString` at least *fails* if it doesn't pass the parser test.